velox/core/SimpleFunctionMetadata.h

```diff
@@ -392,8 +418,22 @@ class UDFHolder final
       return instance_.callAscii(out, args...);
     } else if constexpr (udf_has_call) {
       return instance_.call(out, args...);
-    } else {
+    } else if constexpr (udf_has_callNullable) {
       return instance_.callNullable(out, (&args)...);
+    } else {
+      VELOX_UNREACHABLE(
+          "callAscii should never be called if the UDF only has callNullFree implemented");
```

Comment:
haha we need a diagram that illustrate the control flow of those calls to add sometime.